File: app/src/main/java/com/pixels/blockies/game/game/figures/FigureJ.java
package com.pixels.blockies.game.game.figures;

import com.pixels.blockies.game.game.Grid;

public class FigureJ extends AbstractFigure {

    private int[][][] figures = new int[][][]{
            {
                    {1, Grid.EMPTY, Grid.EMPTY},
                    {1, 1, 1}
            },
            {
                    {1, 1},
                    {1, Grid.EMPTY},
                    {1, Grid.EMPTY}
            },
            {
                    {1, 1, 1},
                    {Grid.EMPTY, Grid.EMPTY, 1}
            },
            {
                    {Grid.EMPTY, 1},
                    {Grid.EMPTY, 1},
                    {1, 1}
            }
    };

    @Override
    protected int getFigureCount() {
        return figures.length;
    }

    @Override
    public int[][] get() {
        return figures[getCurrentRotation()];
    }

    @Override
    public int[][] getNext() {
        return figures[getNextRotation()];
    }
}

File: app/src/main/java/com/pixels/blockies/game/game/figures/FigureJ.kt
package com.pixels.blockies.game.game.figures

import com.pixels.blockies.game.game.Grid

class FigureJ : AbstractFigure() {
    private val figures = arrayOf(
        arrayOf<IntArray?>(
            intArrayOf(1, Grid.Companion.EMPTY, Grid.Companion.EMPTY),
            intArrayOf(1, 1, 1)
        ),
        arrayOf<IntArray?>(
            intArrayOf(1, 1),
            intArrayOf(1, Grid.Companion.EMPTY),
            intArrayOf(1, Grid.Companion.EMPTY)
        ),
        arrayOf<IntArray?>(
            intArrayOf(1, 1, 1),
            intArrayOf(Grid.Companion.EMPTY, Grid.Companion.EMPTY, 1)
        ),
        arrayOf<IntArray?>(
            intArrayOf(Grid.Companion.EMPTY, 1),
            intArrayOf(Grid.Companion.EMPTY, 1),
            intArrayOf(1, 1)
        )
    )

    override fun getFigureCount(): Int {
        return figures.size
    }

    override fun get(): Array<IntArray?> {
        return figures[currentRotation]
    }

    override fun getNext(): Array<IntArray?> {
        return figures[nextRotation]
    }
}